test/parallel/test-timers-args.js

```diff
@@ -3,29 +3,42 @@ require('../common');
 const assert = require('assert');
 
 function range(n) {
-  return 'x'.repeat(n + 1).split('').map(function(_, i) { return i; });
+  return new Array(n).fill(0).map((_, i) => i);
 }
 
 function timeout(nargs) {
   const args = range(nargs);
-  setTimeout.apply(null, [callback, 1].concat(args));
+  const timer = setTimeout(callback, 1, ...args);
 
   function callback() {
-    assert.deepStrictEqual([].slice.call(arguments), args);
+    clearTimeout(timer);
+    assert.deepStrictEqual(Array.from(arguments), args);
     if (nargs < 128) timeout(nargs + 1);
   }
 }
 
 function interval(nargs) {
   const args = range(nargs);
-  const timer = setTimeout.apply(null, [callback, 1].concat(args));
+  const timer = setInterval(callback, 1, ...args);
 
   function callback() {
     clearInterval(timer);
-    assert.deepStrictEqual([].slice.call(arguments), args);
+    assert.deepStrictEqual(Array.from(arguments), args);
     if (nargs < 128) interval(nargs + 1);
   }
 }
 
-timeout(0);
-interval(0);
+function immediate(nargs) {
```

Comment:
IMO this should be in a separate file that is about Immediates rather than Timeouts.

Sure, they are from the same module, but they share pretty much 0 implementation at the current time.